apps/web/src/components/display/UploadFileModal.tsx

```diff
@@ -0,0 +1,96 @@
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+import { Dropzone, onDropType } from '../Dropzone';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTimesCircle } from '@fortawesome/free-solid-svg-icons';
+import Papa from 'papaparse';
+
+import { Modal } from '../Modal';
+import { Button } from '../Button';
+
+export const UploadFileModal: React.FC = () => {
+  const router = useRouter();
+  const isOpen = !!router.query.bulk_upload;
+
+  const handleClose = () => {
+    delete router.query.bulk_upload;
+    router.push(router.route, undefined, { shallow: true });
+  };
+
+  // @todo add confirmation, warning, errors modal
+  return (
+    <Modal open={isOpen} handleClose={handleClose}>
+      <UploadPage closeModal={handleClose} />
+    </Modal>
+  );
+};
+
+/* Upload Page Component */
+interface UploadPageProps {
+  closeModal: () => void;
+}
+
+const UploadPage: React.FC<UploadPageProps> = ({ closeModal }) => {
+  const [file, setFile] = useState<any>(null);
+
+  // parses csv file using papaparse, header uses column headers as keys in JSON data
+  const handleFileUpload = async () => {
+    Papa.parse(file, {
+      header: true,
+      transformHeader: h => {
+        return h.replace(/\s/g, '');
+      },
+      complete: function (results) {
+        console.log('FILE: ', results.data);
```

Comment:
Nit - Maybe add a TODO remove or a comment explaining the context around this console.log